internal/wasm/jit/asm/impl.go

```diff
@@ -0,0 +1,47 @@
+package asm
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math"
+)
+
+type BaseAssemblerImpl struct {
```

Comment:
```suggestion
// BaseAssemblerImpl includes code common to all architectures.
//
// Note: When possible, add code here instead of in architecture-specific files to reduce drift:
// As this is internal, exporting symbols only to reduce duplication is ok.
type BaseAssemblerImpl struct {
```